sql3/planner/opinsert.go

```diff
@@ -264,6 +265,59 @@ func (i *insertRowIter) Next(ctx context.Context) (types.Row, error) {
 						uint64s[i] = uint64(v[i])
 					}
 					row.Values[posVals[idx]] = uint64s
+
+				case []interface{}:
+					// it's a tuple, check length
+					if len(v) != 2 {
+						return nil, sql3.NewErrUnexpectedTimeQuantumTupleLength(0, 0, columnName, rowNumber+1, v, len(v))
+					}
+
+					tupleType, ok := iv.Type().(*parser.DataTypeTuple)
+					if !ok {
+						return nil, sql3.NewErrInternalf("unexpected tuple type '%T'", v[0])
+					}
+
+					// first member must be a timestamp or coercable as one
+					cval, err := coerceValue(tupleType.Members[0], parser.NewDataTypeTimestamp(), v[0], parser.Pos{Line: 0, Column: 0})
```

Comment:
do we not have a valid position we can use here?

also... should the timestamp be the first value? my intuition would have been that the timestamp was the last value in the tuple. did we specify this anywhere? is there a thing we're doing this to be compatible with?

Reply:
Completely arbitrary. If someone can point to prior art where it's the other way, happy to change it.